src/main/java/com/synopsys/integration/blackduck/phonehome/BlackDuckPhoneHomeHelper.java

```diff
@@ -35,35 +34,28 @@ public class BlackDuckPhoneHomeHelper {
+    public static BlackDuckPhoneHomeHelper createPhoneHomeHelper(BlackDuckServicesFactory blackDuckServicesFactory, String apiSecret, String measurementId) {
+        return BlackDuckPhoneHomeHelper.createAsynchronousPhoneHomeHelper(blackDuckServicesFactory, apiSecret, measurementId, new NoThreadExecutorService());
     }
 
-    public static BlackDuckPhoneHomeHelper createAsynchronousPhoneHomeHelper(BlackDuckServicesFactory blackDuckServicesFactory, ExecutorService executorService) {
-        return createAsynchronousPhoneHomeHelper(blackDuckServicesFactory, executorService, GoogleAnalyticsConstants.PRODUCTION_INTEGRATIONS_TRACKING_ID);
-    }
-
-    public static BlackDuckPhoneHomeHelper createAsynchronousPhoneHomeHelper(BlackDuckServicesFactory blackDuckServicesFactory, ExecutorService executorService, String trackingId) {
+    public static BlackDuckPhoneHomeHelper createAsynchronousPhoneHomeHelper(BlackDuckServicesFactory blackDuckServicesFactory, String apiSecret, String measurementId, ExecutorService executorService) {
         BlackDuckRegistrationService blackDuckRegistrationService = blackDuckServicesFactory.createBlackDuckRegistrationService();
 
         IntLogger intLogger = blackDuckServicesFactory.getLogger();
         IntEnvironmentVariables intEnvironmentVariables = blackDuckServicesFactory.getEnvironmentVariables();
         BlackDuckHttpClient blackDuckHttpClient = blackDuckServicesFactory.getBlackDuckHttpClient();
         HttpClientBuilder httpClientBuilder = blackDuckHttpClient.getHttpClientBuilder();
         Gson gson = blackDuckServicesFactory.getGson();
-        PhoneHomeClient phoneHomeClient = BlackDuckPhoneHomeHelper.createPhoneHomeClient(intLogger, httpClientBuilder, gson, trackingId);
+        PhoneHomeClient phoneHomeClient = BlackDuckPhoneHomeHelper.createPhoneHomeClient(intLogger, httpClientBuilder, gson, apiSecret, measurementId);
 
         PhoneHomeService phoneHomeService = PhoneHomeService.createAsynchronousPhoneHomeService(intLogger, phoneHomeClient, executorService);
 
         return new BlackDuckPhoneHomeHelper(intLogger, phoneHomeService, blackDuckRegistrationService, intEnvironmentVariables);
     }
 
-    public static PhoneHomeClient createPhoneHomeClient(IntLogger intLogger, HttpClientBuilder httpClientBuilder, Gson gson) {
-        return createPhoneHomeClient(intLogger, httpClientBuilder, gson, GoogleAnalyticsConstants.PRODUCTION_INTEGRATIONS_TRACKING_ID);
-    }
-
-    public static PhoneHomeClient createPhoneHomeClient(IntLogger intLogger, HttpClientBuilder httpClientBuilder, Gson gson, String trackingId) {
-        return new PhoneHomeClient(intLogger, httpClientBuilder, gson, trackingId);
+    public static PhoneHomeClient createPhoneHomeClient(IntLogger intLogger, HttpClientBuilder httpClientBuilder, Gson gson, String apiSecret, String measurementId) {
+        //return new PhoneHomeClient(intLogger, httpClientBuilder, gson, apiSecret, measurementId);
```

Comment:
I have changed the constructor/API in the phone-home-client to accommodate the GA4 API secret, maintaining the order of parameters used here. Please go ahead and use the new signature

Reply:
updated in https://github.com/blackducksoftware/blackduck-common/pull/417/commits/d9705dba1f58c8364ec53fd84915345c860a74a1